ivy/functional/frontends/numpy/statistics/averages_and_variances.py

```diff
@@ -141,38 +141,42 @@ def std(
     return ret.astype(dtype, copy=False)
 
 
+def weighted_average(
+    a, /, *, axis=None, weights=None, keepdims_kw=None, returned=False
+):
+    if a.shape != weights.shape:
+        weights = ivy.broadcast_to(weights, (a.ndim - 1) * (1,) + weights.shape)
+        weights = weights.swapaxes(-1, axis)
+    weights_sum = weights.sum(axis=axis, **keepdims_kw)
+    mul = ivy.multiply(a, weights)
+    avg = ivy.sum(mul, axis=axis, **keepdims_kw) / weights_sum
+    return avg, weights_sum if returned else avg
+
+
 @to_ivy_arrays_and_back
 @from_zero_dim_arrays_to_scalar
 def average(a, /, *, axis=None, weights=None, returned=False, keepdims=False):
     axis = tuple(axis) if isinstance(axis, list) else axis
-    global avg
-    avg = 0
-
-    if keepdims is None:
-        keepdims_kw = {}
-    else:
-        keepdims_kw = {"keepdims": keepdims}
+    if weights and a.shape != weights.shape and axis is None:
```

Comment:
Incase `a.shape != weights.shape`, we should broadcast `weights` as done in the existing implementation.